test/test_linalg.py

```diff
@@ -2483,8 +2503,7 @@ def test_invariance_error_spectral_decompositions(self, device, dtype):
             Q = torch.linalg.eigh(A).eigenvectors
             Q.sum().abs().backward()
 
-    @skipCUDAIfNoCusolver  # MAGMA backend doesn't work in this case
-    @skipCUDAIfRocm
+    @skipCUDAIfNoCusolverAndNoHipsolver  # MAGMA backend doesn't work in this case
```

Comment:
We don't have anything specific in Python tests for hipBLAS or rocBLAS. Why should we have it for hipSOLVER? Why can't it be so that cuSOLVER == hipSOLVER on the ROCm platform?

Reply:
wrapping the logic from skipCUDAIfNoCusolverAndNoHipsolver into skipCUDAIfNoCusolver